lib/src/Uncertainty/Algorithm/Simulation/CrossEntropyImportanceSampling.cxx

```diff
@@ -0,0 +1,245 @@
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/CrossEntropyImportanceSampling.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class CrossEntropyImportanceSampling
+ */
+
+CLASSNAMEINIT(CrossEntropyImportanceSampling)
+
+static const Factory<CrossEntropyImportanceSampling> Factory_CrossEntropyImportanceSampling;
+
+// Default constructor
+CrossEntropyImportanceSampling::CrossEntropyImportanceSampling()
+  : EventSimulation()
+{
+  // Nothing TO DO
+}
+
+
+// Default constructor
+CrossEntropyImportanceSampling::CrossEntropyImportanceSampling(const RandomVector & event,
+                                                               const Scalar rhoQuantile)
+  : EventSimulation(event)
+  , initialDistribution_(event.getAntecedent().getDistribution())
+{
+  event_ = event;
+  rhoQuantile_ = (event.getOperator()(0, 1) ? rhoQuantile : 1.0 - rhoQuantile);
+}
+
+/* Virtual constructor */
+CrossEntropyImportanceSampling * CrossEntropyImportanceSampling::clone() const
+{
+  return new CrossEntropyImportanceSampling(*this);
+}
+
+// Get rhoQuantile
+Scalar CrossEntropyImportanceSampling::getRhoQuantile() const
+{
+  return rhoQuantile_;
+}
+
+// Set rhoQuantile
+void CrossEntropyImportanceSampling::setRhoQuantile(const Scalar & rhoQuantile)
+{
+  rhoQuantile_ = rhoQuantile;
+}
+
+// Compute Output Samples
+Sample CrossEntropyImportanceSampling::computeOutputSamples(const Sample & ) const
+{
+  throw NotYetImplementedException(HERE) << "In CrossEntropyImportanceSampling::computeOutputSamples(const Sample & inputSamples)";
+}
+
+// Update auxiliary distribution
+void CrossEntropyImportanceSampling::updateAuxiliaryDistribution(const Point & )
+{
+  throw NotYetImplementedException(HERE) << "In CrossEntropyImportanceSampling::updateAuxiliaryDistribution(const Point & auxiliaryDistributionParameters)";
+} 
+  
+// Optimize auxiliary distribution parameters
+Point CrossEntropyImportanceSampling::optimizeAuxiliaryDistributionParameters(const Sample & ) const
+{
+  throw NotYetImplementedException(HERE) << "In CrossEntropyImportanceSampling::optimizeAuxiliaryDistributionParameters(const Sample &)";
+} 
+  
+
+
+// Main function that computes the failure probability
+void CrossEntropyImportanceSampling::run()
+{
+  const UnsignedInteger numberOfSample = getMaximumOuterSampling() * getBlockSize();
```

Comment:
```suggestion
  const UnsignedInteger sampleSize = getMaximumOuterSampling() * getBlockSize();
```
This is a point of statistical / OpenTURNS vocabulary. A `Sample` has a size representing the number of  `Point`s it contains. In this case, there is a single `Sample`, but not several (this is different with the pick-freeze estimator for Sobol' indices, where there are different samples involved in the estimator).

Reply:
Fixed, thanks.